src/components/VmActions/index.js

```diff
@@ -136,12 +137,13 @@ class VmActions extends React.Component {
 
     const vncConsole = vm.get('consoles').find(c => c.get('protocol') === VNC)
     const hasRdp = isWindows(vm.getIn(['os', 'type']))
-    const defaultConsole = vm.get('defaultConsole')
+    const defaultUiConsole = config.get('defaultUiConsole')
     let consoles = []
 
     consoles = vm.get('consoles').map((c) => ({
       priority: 0,
       protocol: c.get('protocol'),
+      uiConsole: toUiConsole(NATIVE, c.get('protocol')),
```

Comment:
This doesn't seem incorrect to me. I think that in case Spice is not supported and vnc is supported then the default ui conosle type should be based on ClientModeVncDefault as well, i.e in case noVnc is supported then this should be the default.

Reply:
The current code provides the requested functionality.  

Note that we always support both VNC flavors if VNC protocol is present:
1.  `NoVnc` is added when `vncConsole` flag is t`true`. 
2.  The `Native` VNC is added by directly mapping the list of available protocols (that's why `toUiConsole()` is used with first parameter set to `NATIVE`).


This logic was inherited from previous versions.

Reply:
@rszwajko the issue raised here is regarding the default console displayed for the VM card in case that the defaultConsole is not supported by that VM.

For reproducing:
set `ClientModeConsoleDefault`=`spice`, `ClientModeVncDefault`=`NoVnc` and  set the VM to support only `VNC`.

Expected result: the default option should be `BrowserVnc`.
Actual result:  the default option is `VNC`.
This seems like not supported by the current code but it was supported by the previous version and by webadmin.

> > > It's not a regression. This fix was introduced in [148f731](https://github.com/oVirt/ovirt-web-ui/commit/148f73159aa72d65f4e5743ecb5d780404efb663)
> > 
> > 
> > Based on webadmin and from current design aspect, we don't really support an option to disable RDP since RDP is always enabled for win vms, even headless ones.
> > We did miss that for older versions and added this functionality back on #1342.
> > So if we want to be compatible with webadmin, we need to add this.
> > Nevertheless since it was not supported till 1.6.8, it's ok for me to leave it as is until someone will complain. Not that important.
> 
> We can fix it under separate issue.

Sure, please open a separate issue for this

Reply:
> @rszwajko the issue raised here is regarding the default console displayed for the VM card in case that the defaultConsole is not supported by that VM.
> 
> For reproducing:
> set `ClientModeConsoleDefault`=`spice`, `ClientModeVncDefault`=`NoVnc` and set the VM to support only `VNC`.
> Expected result: the default option should be BrowserVnc.
> Actual result: the default option is VNC.
> This seems like not supported by the current code but it was supported by the previous version and by webadmin.
>

OK. I understand the use case now. However I'm not sure if it applies to the current design. Previous version and Web Admin provide a single button with black-box style logic (no indication what console will be chosen by the algorithm).
In the new design it's transparent to the user:
1. all available consoles are visible (the default console goes first)
2. the default global console send from the server (server's default console) is displayed in Account Settings (see below - feature available soon). Due to that there is only one such default (we don't have secondary default  for cases like `ClientModeConsoleDefault=spice`, `ClientModeVncDefault=NoVnc` )
3. user can override(completely) the server's default by setting a preferred console in Account Settings (available soon). Once again it's only one value (a limitation of SelectBox UI).
4. user can override the global console by setting per VM value (available in the next phase)

The goal is to have the logic here simple and transparent that's why only 2 levels are used:
1. global - either server's default console or console set by the user in Account Settings
2. per VM (next phase)

If the default is not supported by the given VM (for any reason) then consoles are displayed in default order (as before). Of course we could provide fallback but things will get complex i.e. imagine the case where:
1. ClientModeConsoleDefault`=`spice`, `ClientModeVncDefault`=`NoVnc`
2. preferred console is RDP
3. VM is Linux with VNC only

I'm not convinced if the added value justifies the complexity:) 

![image](https://user-images.githubusercontent.com/64194103/121316816-fe1af400-c909-11eb-987a-274d404d813b.png)